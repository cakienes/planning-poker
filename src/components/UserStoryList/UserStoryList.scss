.userStoryList {
    width: 100%;

    .pageTitle {
        font-size: 22px;
    }
    table {
        margin-top: 20px;
        width: 100%;
        thead {
            background: #afafaf;
        }
        tr {
            width: 100%;

            th,
            td {
                width: 33%;
                border: 1px solid #000;
                padding: 10px 5px;
            }

            &:nth-child(even) {
                background: #ececec;
            }
        }
    }
}
