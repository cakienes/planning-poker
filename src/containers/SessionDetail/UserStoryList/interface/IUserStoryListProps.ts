import ISession from '../../../../interfaces/ISession';

export default interface IUserStoryListProps {
    selectedSession?: ISession;
}
