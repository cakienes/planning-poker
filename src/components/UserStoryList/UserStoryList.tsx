import React from 'react';
import IUserStory from '../../interfaces/IUserStory';
import IUserStoryListProps from './interface/IUserStoryListProps';
import './UserStoryList.scss';

export class UserStoryList extends React.Component<IUserStoryListProps, any> {
    render() {
        return (
            <div className="userStoryList">
                <div className="pageTitle">Story List</div>
                <table>
                    <thead>
                        <tr>
                            <th>Story</th>
                            <th>Story Point</th>
                            <th>Status</th>
                        </tr>
                    </thead>
                    <tbody>{this.renderTableData()}</tbody>
                </table>
            </div>
        );
    }

    renderTableData = (): React.ReactNode => {
        const { selectedSession } = this.props;

        if (selectedSession && selectedSession.userStories.length > 0) {
            return selectedSession.userStories.map((userStory: IUserStory, index: number) => {
                return (
                    <tr key={index}>
                        <td>{userStory.storyName}</td>
                        <td>{userStory.storyPoint}</td>
                        <td>{userStory.status}</td>
                    </tr>
                );
            });
        }
    };
}

export default UserStoryList;
