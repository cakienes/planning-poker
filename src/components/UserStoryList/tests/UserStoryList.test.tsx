import React from 'react';
import { createRenderer, ShallowRenderer } from 'react-test-renderer/shallow';
import IUserStoryListProps from '../interface/IUserStoryListProps';
import UserStoryList from '../UserStoryList';

describe('UserStoryList.tsx', () => {
    let renderer: ShallowRenderer;
    let userStoryListProps: IUserStoryListProps = {};

    beforeEach(() => {
        renderer = createRenderer();

        userStoryListProps = {
            selectedSession: undefined,
        };
    });

    it('renders correctly UserStoryList', async () => {
        expect(renderer.render(<UserStoryList {...userStoryListProps} />)).toMatchSnapshot();
    });

    it('renders correctly UserStoryList while selectedSession.userStories is empty', async () => {
        userStoryListProps = {
            selectedSession: {
                developers: [],
                id: 'id',
                numberOfVoters: 1,
                optionalUrl: undefined,
                sessionName: 'sessionName',
                userStories: [],
            },
        };
        expect(renderer.render(<UserStoryList {...userStoryListProps} />)).toMatchSnapshot();
    });

    it('renders correctly UserStoryList while selectedSession.userStories is not empty', async () => {
        userStoryListProps = {
            selectedSession: {
                developers: [],
                id: 'id',
                numberOfVoters: 1,
                optionalUrl: undefined,
                sessionName: 'sessionName',
                userStories: [
                    {
                        storyName: 'storyName',
                    },
                ],
            },
        };
        expect(renderer.render(<UserStoryList {...userStoryListProps} />)).toMatchSnapshot();
    });
});
