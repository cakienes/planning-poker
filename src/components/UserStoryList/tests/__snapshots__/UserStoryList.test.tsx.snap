// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserStoryList.tsx renders correctly UserStoryList 1`] = `
<div
  className="userStoryList"
>
  <div
    className="pageTitle"
  >
    Story List
  </div>
  <table>
    <thead>
      <tr>
        <th>
          Story
        </th>
        <th>
          Story Point
        </th>
        <th>
          Status
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`UserStoryList.tsx renders correctly UserStoryList while selectedSession.userStories is empty 1`] = `
<div
  className="userStoryList"
>
  <div
    className="pageTitle"
  >
    Story List
  </div>
  <table>
    <thead>
      <tr>
        <th>
          Story
        </th>
        <th>
          Story Point
        </th>
        <th>
          Status
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`UserStoryList.tsx renders correctly UserStoryList while selectedSession.userStories is not empty 1`] = `
<div
  className="userStoryList"
>
  <div
    className="pageTitle"
  >
    Story List
  </div>
  <table>
    <thead>
      <tr>
        <th>
          Story
        </th>
        <th>
          Story Point
        </th>
        <th>
          Status
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          storyName
        </td>
        <td />
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
